provider/k8s.go

```diff
@@ -839,7 +840,7 @@ func (k8s *K8sOfflineStore) sqlTransformation(config TransformationConfig, isUpd
 func (k8s *K8sOfflineStore) checkArgs(args metadata.TransformationArgs) (metadata.KubernetesArgs, error) {
 	k8sArgs, ok := args.(metadata.KubernetesArgs)
 	if !ok {
-		return metadata.KubernetesArgs{}, fmt.Errorf("invalid type used for Kubernetes Arguments")
+		return metadata.KubernetesArgs{}, fferr.NewInvalidArgument(fmt.Errorf("args is not of type KubernetesArgs"))
```

Comment:
Think this should be an internal error since if the struct is wrong then its our fault